### DISCUSSION TITLE
How to work with Multiple Git-Repos

### DISCUSSION THREAD

**Discussion by ig12** [akuity/kargo#2065] [Q&A] [answered]
Hi
Great project - a possible answer to the problems of complexity we face at my company.

We have a Repository structure where different apps belonging to the same platform are stored in their separate repositories.
But the apps are still related to each other, and each new change in any repo needs to be validated. 
As I understand and tested - Kargo supports multiple git repositories.

What I don't understand are the configuration options in the Stage CRD.

I have the following warehouse:
```
apiVersion: kargo.akuity.io/v1alpha1
kind: Warehouse
...
spec:
  subscriptions:
  - image:
      imageSelectionStrategy: SemVer
      repoURL: nginxinc/nginx-unprivileged
      semverConstraint: ^1.25.0
  - git:
      branch: main
      commitSelectionStrategy: NewestFromBranch
      repoURL: https://.../kargo-demo.git
  - git:
      branch: main
      commitSelectionStrategy: NewestFromBranch
      repoURL: https://.../playground/kargo-demo-app2.git
```
and this stage

```
apiVersion: kargo.akuity.io/v1alpha1
kind: Stage
metadata:
  creationTimestamp: "2024-05-23T13:27:27Z"
  finalizers:
  - kargo.akuity.io/finalizer
  generation: 8
  labels:
    kargo.akuity.com/v0.5-compatible: "true"
  name: test
  namespace: kargo-demo
  resourceVersion: "133476"
  uid: 322b2ce0-3d15-418d-a17d-44588767d811
spec:
  promotionMechanisms:
    argoCDAppUpdates:
    - appName: kargo-demo-test
      appNamespace: argocd
    gitRepoUpdates:
    - kustomize:
        images:
        - image: nginxinc/nginx-unprivileged
          path: stages/test
          useDigest: false
      repoURL: https://.../playground/kargo-demo.git
      writeBranch: stage/test
    - kustomize:
        images:
        - image: nginx
          path: base2
          useDigest: false
      repoURL: https://...playground/kargo-demo-app2.git
      writeBranch: stage/test
  subscriptions:
    warehouse: kargo-demo
```

An it works.
But, for the second app, I do not need to update the image, but I could not make it work to promote the commit if the image part was missing. Is there a way to promote without using an image/dummy kustomize image?

Is there an option, or a planned feature to place the two apps in the same repo mut in different paths? the would help to have the deploy clusters only pointing to one repo and to reduce the moving parts.
Because in the actual deployment, I would have some 20 repos to be monitored and promoted

Thanks for you help and insights to the project

**Comment by krancour** [comment, marked as answer]
@ig12 what happens if you leave the entire `kustomize` section out of the second git repo update?

i.e.:

```
    gitRepoUpdates:
    - kustomize:
        images:
        - image: nginxinc/nginx-unprivileged
          path: stages/test
          useDigest: false
      repoURL: https://.../playground/kargo-demo.git
      writeBranch: stage/test
    - repoURL: https://...playground/kargo-demo-app2.git
      writeBranch: stage/test
```

**Comment by ig12** [comment]
Thank you that worked
I can not figure out what I tried but I guess I tried a combination of the two like
```
    gitRepoUpdates:
    - kustomize:
        images:
        - image: nginxinc/nginx-unprivileged
          path: stages/test
          useDigest: false
      repoURL: https://.../playground/kargo-demo.git
      writeBranch: stage/test
    - kustomize:
       repoURL: https://...playground/kargo-demo-app2.git
       writeBranch: stage/test
```
